app/tests/test_domain_commands.py

```diff
@@ -31,3 +43,87 @@ def test_resultCommand_invokeUponSamplePool_resultIsInPool():
     given_command.invoke(given_pool)
 
     assert given_pool.results[given_id] == given_result
+
+
+def test_registerCommand_invokeNoSubproblemBeingComputed_returnsEmptyList():
+    given_id = TestId(42)
+    given_pool = TestPool()
+    given_command = RegisterCommand(given_id)
+
+    register_invoke_output = given_command.invoke(given_pool)
+
+    assert len(register_invoke_output) == 0
+
+
+def test_registerCommand_invokeWithDifferentSubproblemInProgress_returnsEmptyList():
+    given_id = TestId(42)
+    current_id = TestId(16)
+    given_pool = TestPool()
+    given_command = RegisterCommand(given_id)
+    given_pool.in_progress_pool.add(current_id)
```

Comment:
Use SubproblemPool.register(), which places ID in in_progress_pool, but doesn't mark it as currently executed my local machine